rs/index/src/posting_list/file.rs

```diff
@@ -363,9 +354,44 @@ impl<'a> PostingListStorage<'a> for FileBackedAppendablePostingListStorage {
             self.flush_resident_posting_lists_to_disk()?;
         }
 
+        // First write the metadata
+        let metadata_file_access_info = Self::offset_to_file_access_info(
+            self.metadata_current_overall_offset,
+            self.backing_file_size,
+            &self.metadata_mmaps,
+        )?;
+        let metadata_file_offset = metadata_file_access_info.file_offset;
+        let metadata_file_num = metadata_file_access_info.file_num;
+        for (i, mmap) in self.metadata_mmaps.iter().enumerate() {
+            let bytes_to_write = if i == metadata_file_num {
+                metadata_file_offset
+            } else {
+                mmap.len()
+            };
+
+            let bytes_written = wrap_write(writer, &mmap[..bytes_to_write])?;
+            total_bytes_written += bytes_written;
+
+            if bytes_written != bytes_to_write {
+                return Err(anyhow!(
+                    "Failed to write entire metadata mmap: expected {} bytes, wrote {} bytes",
+                    bytes_to_write,
+                    bytes_written
+                ));
+            }
+        }
```

Comment:
similarly, there might be a way to reuse the code between metadata and posting lists?